src/Domain/Creators/ExpandOverBlocks.cpp

```diff
@@ -0,0 +1,53 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/Creators/ExpandOverBlocks.hpp"
+
+#include <array>
+#include <cstddef>
+#include <exception>
+#include <string>
+#include <vector>
+
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeArray.hpp"
+
+namespace domain {
+
+template <typename T, size_t Dim>
+std::vector<std::array<T, Dim>> ExpandOverBlocks<T, Dim>::operator()(
+    const T& value) const {
+  return {num_blocks, make_array<Dim>(value)};
+}
+
+template <typename T, size_t Dim>
+std::vector<std::array<T, Dim>> ExpandOverBlocks<T, Dim>::operator()(
+    const std::array<T, Dim>& value) const {
+  return {num_blocks, value};
+}
+
+template <typename T, size_t Dim>
+std::vector<std::array<T, Dim>> ExpandOverBlocks<T, Dim>::operator()(
+    const std::vector<std::array<T, Dim>> value) const {
+  if (value.size() != num_blocks) {
+    throw std::length_error{"You supplied " + std::to_string(value.size()) +
```

Comment:
Maybe add a note that this is throwing instead of `ERROR`ing to be compatible with the option parser?